code/modules/core_implant/cruciform/rituals/construction.dm

```diff
@@ -0,0 +1,246 @@
+	user.visible_message(SPAN_NOTICE("You see as [user] passes his hands over something."),SPAN_NOTICE("You see your faith take physical form as you concentrate on [blueprint.name] image"))
+	if(!do_after(user, 5 SECONDS, target_turf))
+		fail("You feel something is judging you upon your impatience",user,C,targets)
+		return
+	if(!items_check(user, target_turf, blueprint))
+		fail("Something got stolen!",user,C,targets)
+		return
+	//magic has to be a bit innacurate
+
+	for(var/item_type in blueprint.materials)
+		var/t = locate(item_type) in target_turf.contents
+		qdel(t)
+		
+	user.visible_message(SPAN_NOTICE("You hear a soft humming sound as [user] finishes his ritual."),SPAN_NOTICE("You take a deep breath as the divine manifestation finishes."))
+	var/build_path = blueprint.build_path
+	new build_path(target_turf)	
+
+/datum/ritual/cruciform/priest/construction/proc/items_check(mob/user,turf/target, datum/nt_blueprint/blueprint)
+	var/list/turf_contents = target.contents
+	
+	for(var/item_type in blueprint.materials)
+		var/located_raw = locate(item_type) in turf_contents
+		//single item check is handled there, rest of func is for stacked items or items with containers
+		if(!located_raw)
+			return FALSE
+
+		var/required_amount = blueprint.materials[item_type]
+        // I hope it is fast enough
+        // could have initialized it in glob
+		if(item_type in typesof(/obj/item/stack/))
+			var/obj/item/stack/stacked = located_raw
+			if(stacked.amount < required_amount)
+				return FALSE
+	return TRUE
+/datum/nt_blueprint/
+	var/name = "Report me"
+	var/build_path
+	var/list/materials
+	
+/datum/nt_blueprint/canister
+	name = "Biomatter Canister"
+	build_path = /obj/structure/reagent_dispensers/biomatter
+	materials = list(
+		/obj/item/stack/material/steel = 8,
+		/obj/item/stack/material/plastic = 2
+	)
+/datum/nt_blueprint/canister/large  
+	name = "Large Biomatter Canister"
+	build_path = /obj/structure/reagent_dispensers/biomatter/large
+	materials = list(
+		/obj/item/stack/material/steel = 16,
+		/obj/item/stack/material/plastic = 4,
+		/obj/item/stack/material/plasteel = 2,
+	)
+
+/datum/nt_blueprint/machinery
+
+/datum/nt_blueprint/machinery/obelisk
+	name = "Obelisk"
+	build_path = /obj/machinery/power/nt_obelisk
+	materials = list(
+		/obj/item/stack/material/plasteel = 10,
+		/obj/item/stack/material/gold = 5,
+		/CRUCIFORM_TYPE = 1
+	)
+/datum/nt_blueprint/machinery/bioprinter
+	name = "Biomatter Printer"
+	build_path = /obj/machinery/autolathe/bioprinter
+	materials = list(
+		/obj/item/stack/material/steel = 10,
+		/obj/item/stack/material/glass = 2,
+		/obj/item/stack/material/silver = 6,
+		/obj/item/weapon/storage/toolbox = 1
+	)
+
+/datum/nt_blueprint/machinery/solidifier
+	name = "Biomatter Solidifier"
+	build_path = /obj/machinery/biomatter_solidifier
+	materials = list(
+		/obj/item/stack/material/steel = 20,
+		/obj/item/stack/material/silver = 5,
+		/obj/structure/reagent_dispensers/biomatter = 1
```

Comment:
Is there even a way to craft these at the moment?

If not; I think you should do one of the following:
- make these craftable
- use something that can be crafted so that NT can set up a "factory" without being limited to only their starting containers. 

This would also prevent a situation where all the barrels are converted to solidifiers.

Reply:
`/datum/nt_blueprint/canister` and `/datum/nt_blueprint/canister/large`